provider/coredns/coredns.go

```diff
@@ -148,10 +176,45 @@ func (c etcdClient) DeleteService(ctx context.Context, key string) error {
 	ctx, cancel := context.WithTimeout(ctx, etcdTimeout)
 	defer cancel()
 
+	if owned, err := c.isOwnedBy(ctx, key); err != nil {
+		return err
+	} else if !owned {
+		return fmt.Errorf("key %q is not owned by this service", key)
+	}
+
 	_, err := c.client.Delete(ctx, key, etcdcv3.WithPrefix())
 	return err
 }
 
+func (c etcdClient) isOwnedBy(ctx context.Context, key string) (bool, error) {
+	if !c.strictlyOwned {
+		return true, nil
+	}
+
+	r, err := c.client.Get(ctx, key)
+	switch {
+	case err != nil:
+		return false, err
+	case r == nil:
+		return true, nil
+	case len(r.Kvs) > 1:
+		return false, fmt.Errorf("found multiple keys with the same key this service")
```

Comment:
please try to rephrase, I don't really understand the error message. I can guess but you can make this better

Reply:
rephrased